src/main/java/yanagishima/servlet/HiveServlet.java

```diff
@@ -3,97 +3,85 @@
 import static java.lang.String.format;
 import static yanagishima.util.AccessControlUtil.sendForbiddenError;
 import static yanagishima.util.AccessControlUtil.validateDatasource;
-import static yanagishima.util.HttpRequestUtil.getOrDefaultParameter;
-import static yanagishima.util.HttpRequestUtil.getRequiredParameter;
-import static yanagishima.util.JsonUtil.writeJSON;
 
-import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
-import javax.inject.Inject;
-import javax.inject.Singleton;
-import javax.servlet.ServletException;
-import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import yanagishima.config.YanagishimaConfig;
 import yanagishima.exception.HiveQueryErrorException;
 import yanagishima.model.hive.HiveQueryResult;
 import yanagishima.service.HiveService;
 
 @Slf4j
-@Singleton
-public class HiveServlet extends HttpServlet {
-    private static final long serialVersionUID = 1L;
-
+@RestController
+@RequiredArgsConstructor
+public class HiveServlet {
     private final YanagishimaConfig config;
     private final HiveService hiveService;
 
-    @Inject
-    public HiveServlet(YanagishimaConfig config, HiveService hiveService) {
-        this.config = config;
-        this.hiveService = hiveService;
-    }
-
-    @Override
-    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        Map<String, Object> reponseBody = new HashMap<>();
+    @PostMapping(path = {"hive", "spark"})
+    public Map<String, Object> post(@RequestParam String datasource,
+                                    @RequestParam String engine,
+                                    @RequestParam(name = "user", required = false) Optional<String> hiveUser,
+                                    @RequestParam(name = "password", required = false) Optional<String> hivePassword,
+                                    @RequestParam(defaultValue = "false") boolean store,
```

Comment:
Are there any reasons not to use ```@RequestParam(required = false) String query```  like ```HiveAsyncServlet``` ?

Reply:
I was just missing it. Fixed.

Reply:
`@RequestParam(required = false) String query`?
not `@RequestParam(defaultValue = "false") String query`

Reply:
Sorry, fixed again.